cmd/dep/status.go

```diff
@@ -722,6 +732,234 @@ outer:
 	return hasMissingPkgs, 0, errInputDigestMismatch
 }
 
+func runOld(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceManager) (hasMissingPkgs bool, errCount int, err error) {
+	ptree, err := p.ParseRootPackageTree()
+	if err != nil {
+		return false, 0, err
+	}
+
+	// Solve the dependencies so we can check if the current dependencies are old
+	params := gps.SolveParameters{
+		ProjectAnalyzer: dep.Analyzer{},
+		RootDir:         p.AbsRoot,
+		RootPackageTree: ptree,
+		Manifest:        p.Manifest,
+		ChangeAll:       true,
+	}
+	s, err := gps.Prepare(params, sm)
+	if err != nil {
+		return false, 0, errors.Wrapf(err, "could not set up solver for input hashing")
+	}
+
+	if err := out.BasicHeader(); err != nil {
```

Comment:
We need a different struct for old status. `BasicHeader` contains too much details that we don't need.
If you see https://github.com/golang/dep/issues/1383#issuecomment-347788937 , that's kind of what we need.